### PR TITLE
Add cache

### PR DESCRIPTION
**Pull request by jean2262**
address #24

### CODE REVIEW

File: cdsodatacli/query.py [hunk 1]
```diff
@@ -42,22 +50,23 @@ def fetch_data(gdf=None, date=None, dtime=None, timedelta_slice=None, start_date
...
-    if min_sea_percent is not None:
-        full_data = sea_percent(collected_data=full_data, min_sea_percent=min_sea_percent)
+        # Convert all Multipolygon to Polygon and add geometry as new column
+        full_data = multi_to_poly(collected_data=data_dedup)
+        if min_sea_percent is not None:
+            full_data = sea_percent(collected_data=full_data, min_sea_percent=min_sea_percent)
 
-    if fig is not None:
-        fig(collected_data=full_data)
+        if figure:
```
Comment (agrouaze) [inline cdsodatacli/query.py:68]: I am not fan with function to do figures in this library. I propose to remove it.

File: cdsodatacli/query.py [hunk 2]
```diff
@@ -279,126 +288,84 @@ def urls_creat(gdf, top=None):
...
+            collected_data = pd.concat([collected_data, df])
+            pbar.update(1)
+    return collected_data
 
 
-def remove_duplicates(safes_ori, keep_list=[]):
-    """
-    Remove duplicate safe (ie same footprint with same date, but different prodid)
-    """
+def remove_duplicates(safes):
```
Comment (agrouaze) [inline cdsodatacli/query.py:332]: could you add docstrings to your methods?

**Review by agrouaze — CHANGES_REQUESTED** [review]
user should be able to choose where to store the cache files.

File: cdsodatacli/query.py (hunk 1 above)
Comment (jean2262) [reply cdsodatacli/query.py:87]: So you mean we don't need to show the figure?

File: cdsodatacli/query.py (hunk 2 above)
Comment (jean2262) [reply cdsodatacli/query.py:356]: Alright

File: cdsodatacli/query.py [hunk 3]
```diff
@@ -261,144 +291,105 @@ def urls_creat(gdf, top=None):
...
     processing_time = end_time - start_time
-    print(f"multi_to_poly processing time:{processing_time}s")
+    logging.debug(f"multi_to_poly processing time: {processing_time}s")
     return collected_data
 
 
 def sea_percent(collected_data, min_sea_percent=None):
     start_time = time.time()
-    warnings.simplefilter(action='ignore', category=FutureWarning)
+    warnings.filterwarnings("ignore", category=UserWarning)
```
Comment (agrouaze) [inline cdsodatacli/query.py:383]: it is dangerous to hide warnings, much better to clean the code.

**Pull request merged**